### ISSUE TITLE
Type 'T' does not satisfy the constraint '"url"' when using type parameter inside 'Pick'

### ISSUE THREAD

**Issue by LukasLSC** [Microsoft/TypeScript#30446]
**TypeScript Version:**  3.4.0-dev.201xxxxx

**Search Terms:** Type 'T' does not satisfy the constraint '"url"', Type 'T' does not satisfy the constraint.

**Code**
```ts
interface FormField {
    url: number;
}

// Error `Type 'T' does not satisfy the constraint '"url"'`
interface FormFieldMinMax<T = 'url'> {
    min: Pick<FormField, T>;
}

// No errors
interface FormFieldMinMax {
    min: Pick<FormField, 'url'>;
}
```

**Expected behavior:** No errors.

**Actual behavior:** An error `Type 'T' does not satisfy the constraint '"url"'` is shown.

**Playground Link:**  [playground](https://www.typescriptlang.org/play/#src=interface%20FormField%20%7B%0D%0A%20%20%20%20url%3A%20number%3B%0D%0A%7D%0D%0A%0D%0Ainterface%20FormFieldMinMax%3CT%20%3D%20'url'%3E%20%7B%0D%0A%20%20%20%20min%3A%20Pick%3CFormField%2C%20T%3E%3B%0D%0A%7D%0D%0A%0D%0Ainterface%20FormFieldMinMax%20%7B%0D%0A%20%20%20%20min%3A%20Pick%3CFormField%2C%20'url'%3E%3B%0D%0A%7D).

**Comment by tadhgmister** [comment]
The constraint it is referring to is that the second argument to Pick must be a subset of the keys of the first argument. In this case your generic type T has no constraints so this would be valid:

`FormFieldMinMax<{a:number}>`

However you obviously can't pick `{a:number}` from the fields of FormField since that makes no sense.
You need to constrain T to be a subset of the keys in FormField:

```
interface FormFieldMinMax<T extends keyof FormField= 'url'> {
    min: Pick<FormField, T>;
}
```

**Comment by LukasLSC** [comment]
Yea, it's a weird thing I didn't think about, thanks 👍
